src/lib/query/contract.ts

```diff
@@ -1,5 +1,56 @@
 import { graphql } from "lib/gql";
 
+export const getContractByContractAddressQueryDocument = graphql(`
+  query getContractByContractAddressQueryDocument($contractAddress: String!) {
+    contracts_by_pk(address: $contractAddress) {
+      address
+      code_id
+      label
+      accountByInitBy {
+        address
+      }
+      admin: account {
+        address
+      }
+    }
+  }
+`);
+
+export const getInstantiateDetailByContractQueryDocument = graphql(`
+  query getInstantiateDetailByContractQueryDocument($contractAddress: String!) {
+    contracts_by_pk(address: $contractAddress) {
+      init_msg
+      transaction {
+        hash
+        block_height
+        block {
+          timestamp
+        }
+      }
+      contract_proposals(
+        where: {
+          proposal: {
+            type: {
+              _in: [
+                "InstantiateContract"
+                "InstantiateContract2"
+                "SoftwareUpgrade"
+              ]
+            }
+          }
+        }
+        order_by: { proposal: { id: asc } }
+        limit: 1
```

Comment:
Could you also talk with the back-end team to ensure the correctness of proposal query?

Reply:
this is correct